### ISSUE TITLE
unable to create kms key

### ISSUE THREAD

**Issue by madhureddy08** [crossplane/provider-aws#1056]
```
kubectl apply -f - <<EOF
apiVersion: kms.aws.crossplane.io/v1alpha1
kind: Key
metadata:
  name: testkmskey
spec:
  forProvider:
    policy: |-
      {
        "Version": "2012-10-17",
        "Statement": [
          {
            "Sid": "Enable IAM User Permissions"
            "Effect": "Allow",
            "Principal": {
              "AWS": "arn:aws:iam::XXXX:user/root"
            },
            "Action": "kms:*",
            "Resource": "*"
          }
        ]
      }
    region: us-east-2
    description: test
EOF
```

When tried creating the kmskey the output says **created** 
but when trying to describe the key result is 
**observe failed: failed to describe Key: NotFoundException: Invalid keyId testkmskey**

**Comment by haarchri** [comment]
Did you test with master branch ?

**Comment by madhureddy08** [comment]
```
kubectl apply -f - <<EOF
apiVersion: kms.aws.crossplane.io/v1alpha1
kind: Key
metadata:
  name: testkmskey
spec:
  deletionPolicy: Delete
  forProvider:
    region: us-east-2
    pendingWindowInDays: 21
    tags:
      - tagKey: app
        tagValue: test
EOF
```

**Comment by haarchri** [comment]
Can you post full kubectl describe output for your key?

**Comment by madhureddy08** [comment]
bash-3.2$ kubectl describe Key testkmskey
Name:         testkmskey
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-name: testkmskey
API Version:  kms.aws.crossplane.io/v1alpha1
Kind:         Key
Metadata:
  Creation Timestamp:  2022-01-07T20:51:49Z
  Generation:          2
  Managed Fields:
    API Version:  kms.aws.crossplane.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:crossplane.io/external-name:
      f:spec:
        f:providerConfigRef:
          .:
          f:name:
      f:status:
        .:
        f:atProvider:
        f:conditions:
    Manager:      crossplane-aws-provider
    Operation:    Update
    Time:         2022-01-07T20:51:49Z
    API Version:  kms.aws.crossplane.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:deletionPolicy:
        f:forProvider:
          .:
          f:pendingWindowInDays:
          f:region:
          f:tags:
    Manager:         kubectl-client-side-apply
    Operation:       Update
    Time:            2022-01-07T20:51:49Z
  Resource Version:  37289912
  UID:               1089ea43-554a-44d9-ab46-53f51113f061
Spec:
  Deletion Policy:  Delete
  For Provider:
    Pending Window In Days:  21
    Region:                  us-east-2
    Tags:
      Tag Key:    app
      Tag Value:  test
  Provider Config Ref:
    Name:  default
Status:
  At Provider:
  Conditions:
    Last Transition Time:  2022-01-07T20:51:49Z
    Message:               observe failed: failed to describe Key: NotFoundException: Invalid keyId testkmskey
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
Events:
  Type     Reason                         Age                From                               Message
  ----     ------                         ----               ----                               -------
  Warning  CannotObserveExternalResource  28s (x3 over 28s)  managed/key.kms.aws.crossplane.io  failed to describe Key: NotFoundException: Invalid keyId testkmskey``

**Comment by haarchri** [comment]
tested with latest master:

```
➜  provider-aws git:(master) kubectl apply -f - <<EOF
apiVersion: kms.aws.crossplane.io/v1alpha1
kind: Key
metadata:
  name: testkmskey
spec:
  deletionPolicy: Delete
  forProvider:
    region: us-east-2
    pendingWindowInDays: 21
    tags:
      - tagKey: app
        tagValue: test
  providerConfigRef:
    name: example
EOF
key.kms.aws.crossplane.io/testkmskey created
```

```
➜  provider-aws git:(master) kubectl describe key.kms.aws.crossplane.io/testkmskey
Name:         testkmskey
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-create-pending: 2022-01-07T22:00:18+01:00
              crossplane.io/external-create-succeeded: 2022-01-07T22:00:19+01:00
              crossplane.io/external-name: 652e1d2a-d657-461d-a18e-517eed292c02
API Version:  kms.aws.crossplane.io/v1alpha1
Kind:         Key
Metadata:
  Creation Timestamp:  2022-01-07T21:00:18Z
  Finalizers:
    finalizer.managedresource.crossplane.io
  Generation:  2
  Resource Version:  2901872
  UID:               30308070-9ba0-4ade-9411-9d38ee1cf0b8
Spec:
  Deletion Policy:  Delete
  For Provider:
    Enabled:                 true
    Pending Window In Days:  21
    Policy:                  {
  "Version" : "2012-10-17",
  "Id" : "key-default-1",
  "Statement" : [ {
    "Sid" : "Enable IAM User Permissions",
    "Effect" : "Allow",
    "Principal" : {
      "AWS" : "arn:aws:iam::255932642927:root"
    },
    "Action" : "kms:*",
    "Resource" : "*"
  } ]
}
    Region:  us-east-2
    Tags:
      Tag Key:    app
      Tag Value:  test
  Provider Config Ref:
    Name:  example
Status:
  At Provider:
    Arn:             arn:aws:kms:us-east-2:255932642927:key/652e1d2a-d657-461d-a18e-517eed292c02
    Aws Account ID:  255932642927
    Creation Date:   2022-01-07T21:00:19Z
    Enabled:         true
    Encryption Algorithms:
      SYMMETRIC_DEFAULT
    Key ID:       652e1d2a-d657-461d-a18e-517eed292c02
    Key Manager:  CUSTOMER
    Key State:    Enabled
  Conditions:
    Last Transition Time:  2022-01-07T21:00:21Z
    Reason:                Available
    Status:                True
    Type:                  Ready
    Last Transition Time:  2022-01-07T21:00:19Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
Events:
  Type     Reason                           Age                From                               Message
```

**Comment by madhureddy08** [comment]
I installed crossplane from master using the below commands.

kubectl create namespace crossplane-system
helm repo add crossplane-master https://charts.crossplane.io/master/
helm repo update
helm install crossplane --namespace crossplane-system crossplane-master/crossplane

Ran the same code. But still getting same error

**Comment by madhureddy08** [comment]
Name:         testkmskey
Namespace:    
Labels:       <none>
Annotations:  crossplane.io/external-name: testkmskey
API Version:  kms.aws.crossplane.io/v1alpha1
Kind:         Key
Metadata:
  Creation Timestamp:  2022-01-10T18:16:25Z
  Generation:          2
  Managed Fields:
    API Version:  kms.aws.crossplane.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:deletionPolicy:
        f:forProvider:
          .:
          f:pendingWindowInDays:
          f:region:
          f:tags:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2022-01-10T18:16:25Z
    API Version:  kms.aws.crossplane.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          f:crossplane.io/external-name:
      f:spec:
        f:providerConfigRef:
          .:
          f:name:
      f:status:
        .:
        f:atProvider:
        f:conditions:
    Manager:         crossplane-aws-provider
    Operation:       Update
    Time:            2022-01-10T18:16:26Z
  Resource Version:  11581
  UID:               139b11c7-2282-4871-84de-7ff09a4257cc
Spec:
  Deletion Policy:  Delete
  For Provider:
    Pending Window In Days:  21
    Region:                  us-east-1
    Tags:
      Tag Key:    app
      Tag Value:  test
  Provider Config Ref:
    Name:  default
Status:
  At Provider:
  Conditions:
    Last Transition Time:  2022-01-10T18:16:26Z
    Message:               observe failed: failed to describe Key: NotFoundException: Invalid keyId testkmskey
    Reason:                ReconcileError
    Status:                False
    Type:                  Synced
Events:
  Type     Reason                         Age                From                               Message
  ----     ------                         ----               ----                               -------
  Warning  CannotObserveExternalResource  8s (x25 over 19m)  managed/key.kms.aws.crossplane.io  failed to describe Key: NotFoundException: Invalid keyId testkmskey
``

**Comment by haarchri** [comment]
which provider-aws you using ? in < v0.23.0 / master we had problem and fixed it for upcomig v0.23.0 Release

please post output kubectl get pkgrev

**Comment by madhureddy08** [comment]
```
bash-3.2$ kubectl get pkgrev
NAME                                                                               HEALTHY   REVISION   IMAGE                                                    STATE    DEP-FOUND   DEP-INSTALLED   AGE
configurationrevision.pkg.crossplane.io/xp-getting-started-with-aws-a0ed9a90578a   True      1          registry.upbound.io/xp/getting-started-with-aws:v1.5.1   Active   1           1               39d

NAME                                                                      HEALTHY   REVISION   IMAGE                             STATE    DEP-FOUND   DEP-INSTALLED   AGE
providerrevision.pkg.crossplane.io/crossplane-provider-aws-9c9ba245db19   True      1          crossplane/provider-aws:v0.18.2   Active                               39d
```

**Comment by haarchri** [comment]
You can see that you running provider-aws:v0.18.2 the issue is fixed on provider-aws > v0.22.0 / master
